### PR TITLE
Lint YAML Config Files before processing

### PR DESCRIPTION
**Pull request by QMalcolm**
There have been issues with YAML config files such as missing "---" dividers between top level object. That specific issue meant that if there were multiple metrics defined in a file, but missing the "---" dividers between them, all the metrics would be silently swallowed except the last specified in the file. The new errors people will see when validating their configs will look like the following:
```
✖ Breaking issues found in config yaml files
• ERROR: in file `opportunities.yaml` on line #84 - duplication of key "metric" in mapping. Key "metric" is a top level object. If multiple top level objects are specified in the same file, they must be separated using "---"
```

### CODE REVIEW

**Comment by QMalcolm** [comment]
The tests failing during the unit test github action aren't reproducing locally. This might take a moment to debug.

File: metricflow/model/parsing/config_linter.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+from metricflow.model.validations.validator_helpers import ValidationContext, ValidationIssue, ValidationIssueLevel
+
+WARNING = "warning"
+ERROR = "error"
+DISABLE = "disable"
+LEVEL = "level"
+
+
+class ConfigLinter:  # noqa: D
```
Comment (plypaul) [inline metricflow/model/parsing/config_linter.py:15]: Can you add a description?

File: metricflow/cli/main.py [hunk 2]
```diff
@@ -695,6 +697,19 @@ def validate_configs(cfg: CLIContext, dw_timeout: Optional[int] = None, skip_dw:
     """Perform validations against the defined model configurations."""
     cfg.verbose = True
 
+    lint_spinner = Halo(text="Checking for yaml lint struture issues", spinner="dots")
```
Comment (plypaul) [inline metricflow/cli/main.py:700]: Nit: Maybe `Checking for YAML format issues`, `Checking for YAML lint issues`, etc.

File: metricflow/model/parsing/config_linter.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+            rules.new_line_at_end_of_file.ID: DISABLE,
+            rules.new_lines.ID: DISABLE,
+            rules.octal_values.ID: DISABLE,  # TODO: Turn on as warning once warnings are less noisy in cli
+            rules.quoted_strings.ID: DISABLE,
+            rules.trailing_spaces.ID: DISABLE,  # TODO: Turn on as warning once warnings are less noisy in cli
+            rules.truthy.ID: DISABLE,
+        },
+    }
+
+    def __init__(self, lint_config: Optional[str] = None) -> None:  # noqa: D
```
Comment (plypaul) [inline metricflow/model/parsing/config_linter.py:44]: Can you point to where a developer can find the format for the `lint_config`?

**Review by plypaul — COMMENTED** [review]
Can you add a test for a lint failure?

File: metricflow/cli/main.py [hunk 4]
```diff
@@ -695,6 +697,19 @@ def validate_configs(cfg: CLIContext, dw_timeout: Optional[int] = None, skip_dw:
     """Perform validations against the defined model configurations."""
     cfg.verbose = True
 
+    lint_spinner = Halo(text="Checking for YAML format issues", spinner="dots")
+    lint_spinner.start()
+
+    lint_issues = ConfigLinter().lint_dir(path_to_models(handler=cfg.config))
+    lint_errors = _filter_issues(lint_issues, [ValidationIssueLevel.ERROR, ValidationIssueLevel.FATAL])
+    if not lint_errors:
+        lint_spinner.succeed("🎉 Successfully linted config yaml files")
```
Comment (plypaul) [inline metricflow/cli/main.py:706]: Nit: YAML

File: metricflow/cli/main.py [hunk 5]
```diff
@@ -695,6 +697,19 @@ def validate_configs(cfg: CLIContext, dw_timeout: Optional[int] = None, skip_dw:
...
+        _print_issues(lint_issues)
+    else:
+        lint_spinner.fail("Breaking issues found in config yaml files")
```
Comment (plypaul) [inline metricflow/cli/main.py:709]: Also YAML

File: metricflow/test/cli/test_cli.py [hunk 6]
```diff
@@ -97,22 +99,34 @@ def test_validate_configs_data_warehouse_validations(cli_runner: MetricFlowCliRu
         ValidationError(context=None, message="Data Warehouse Error"),  # type: ignore
     ]
 
-    with patch.object(CLIContext, "sql_client", return_value=None):  # type: ignore
-        with patch("metricflow.cli.main._run_dw_validations", return_value=dw_validation_issues):
-            resp = cli_runner.run(validate_configs)
+    with patch("metricflow.cli.main.path_to_models", return_value=""):
+        with patch.object(CLIContext, "sql_client", return_value=None):  # type: ignore
```
Comment (plypaul) [inline metricflow/test/cli/test_cli.py:103]: Context behind this patch? Seems off since the types won't match up.

**Review by plypaul — APPROVED** [review]
Nice updates! Left some minor comments though.

**Pull request merged**